### ISSUE TITLE
[feature request] Set viewBox Bounds during SVG export

### ISSUE THREAD

**Issue by davidsara** [archimatetool/archi-scripting-plugin#99]
Hi!
Current API allows to export to SVG using $.model.renderViewToSVG(view, filePath, setViewBox)

Would it be possible to add viewBoxBounds optional parameter to enter min_x, min_y, width and height as seen in GUI?

![image](https://user-images.githubusercontent.com/20066733/139021294-3e0bb18e-a89c-4cf3-a614-16b97fe8a211.png)

For example:
``` js
var viewBoxBounds = {min_x: 0, min_y: 0, width: 1000, height: 495};
$.model.renderViewToSVG(view, "myView.svg", "true", viewBoxBounds);
```

Thanks, David

**Comment by Phillipus** [comment]
I'll look at it. It requires a change to Archi itself as well as jArchi so would have to tie in with the next Archi release.

**Comment by davidsara** [comment]
Just FYI, for now I'm using this simple Python script to set minimum width of exported SVG files. Could be useful workaround.

```python
from bs4 import BeautifulSoup
import os
import os.path


def setViewBox(fileName):
    with open(fileName, 'r') as f:
        data = f.read()

    svg_xml = BeautifulSoup(data, 'xml')
    svg = svg_xml.find('svg')
    viewBox = svg.get('viewBox')
    values = viewBox.split()
    if (int(values[2]) < 1200):
        svg['viewBox'] = values[0] + ' ' + values[1] + ' 1200 ' + values[3]
        with open(fileName, 'w') as f:
            f.write(str(svg_xml))


for dirpath, dirnames, filenames in os.walk("."):
    for filename in [f for f in filenames if f.endswith(".svg")]:
        setViewBox(os.path.join(dirpath, filename))
```

**Comment by Phillipus** [comment]
This will be implemented as part of the new API with options. You'll be able to set an option like:

```js
var options = {setViewBox: true, viewBoxBounds: "0 0 1000 495"};
$.model.renderViewToSVG(view, "pathTo/test.svg", options);
```

See https://github.com/archimatetool/archi-scripting-plugin/issues/130#issuecomment-2295300342

**Comment by Phillipus** [comment]
This is implemented in jArchi 1.7.
